src/components/Table/TasksTable.tsx

```diff
@@ -0,0 +1,179 @@
+import { Table } from "antd";
+import { ColumnProps } from "antd/es/table";
+import { SortOrder as antSortOrder } from "antd/lib/table/interface";
+import { ErrorBoundary } from "components/ErrorBoundary";
+import { StyledRouterLink } from "components/styles";
+import { TaskStatusBadge } from "components/TaskStatusBadge";
+import { WordBreak } from "components/Typography";
+import { getTaskRoute } from "constants/routes";
+import {
+  TaskResult,
+  SortDirection,
+  TaskSortCategory,
+  SortOrder,
+} from "gql/generated/types";
+import { TableOnChange } from "types/task";
+
+interface TasksTableProps {
+  tasks: any;
+  tableChangeHandler?: TableOnChange<TaskResult>;
+  onExpand?: (expanded: boolean) => void;
+  onClickTaskLink?: (taskId: string) => void;
+  sorts?: SortOrder[];
+}
+export const TasksTable: React.FC<TasksTableProps> = ({
+  tasks,
+  tableChangeHandler,
+  onExpand,
+  onClickTaskLink,
+  sorts,
+}) => (
+  <Table
+    data-test-id="tasks-table"
+    rowKey={rowKey}
+    pagination={false}
+    columns={
+      sorts
+        ? getColumnDefsControlled(sorts, onClickTaskLink)
+        : getColumnDefs(onClickTaskLink)
+    }
+    dataSource={tasks}
+    onChange={tableChangeHandler}
+    childrenColumnName="executionTasksFull"
+    expandable={{
+      onExpand: (expanded) => {
+        onExpand(expanded);
+      },
+    }}
+  />
+);
+
+const getColumnDefs = (onClickTaskLink): ColumnProps<TaskResult>[] => [
```

Comment:
there some repeated key value pairs between the two getColumn helper functions. can we combine `title, key, dataIndex, width` and expand them within the two functions